### PR TITLE
Fix golint failures in some pkg/kubelet packages

### PR DESCRIPTION
**Pull request by obitech**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes golint failures for packages under `pkg/kubelet/`:

```
pkg/kubelet/pod
pkg/kubelet/metrics
pkg/kubelet/configmap
pkg/kubelet/config
```
ref: #68026

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/config/apiserver.go
Comment (mattjmcnaughton) [inline pkg/kubelet/config/apiserver.go]: Curiosity question... how did this comment violate the golinter? Or did you just decide it wasn't necessary?

Either is fine, just curious :)

File: pkg/kubelet/metrics/metrics.go
Comment (mattjmcnaughton) [inline pkg/kubelet/metrics/metrics.go]: I'm a little worried about any discrepancies between these variable comments and the `Help` text. For example, for `NodeName`, should our comment instead be `// NodeName is a Gauge that tracks the node's name. The count is always 1.`?

**Review by mattjmcnaughton — APPROVED** [review]
Overall, this looks good to me (modulo some small changes which I think should be easy)! I don't have approval rights, but thanks for tackling these lint errors :)

One request - can you please also update `.hack/.golint_failures` to remove these packages (if they are now fully passing golint)?

**Comment by mattjmcnaughton** [comment]
@obitech thanks for your pr :) After making the changes I described, you may want to tag an additional approver. You can find a list of approvers here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/OWNERS

**Comment by mattjmcnaughton** [comment]
Just a heads up, the `pull-kubernetes-verify` test is raising legitimate errors that we'll want to address before this is merged.

**Comment by obitech** [comment]
Hey Matt,

thanks for your reply on this! I will check out your suggestions. Regarding the `pull-kubernetes` tests, I'll fix it :)

File: pkg/kubelet/config/apiserver.go
Comment (obitech) [reply pkg/kubelet/config/apiserver.go]: Hmm, when I first ran it, golint gave me an error because every package should only have one package level comment (basically a comment above the `package` statement). However, if I try to reproduce it (by putting comments back into the top of the files) now, I don't get  any errors. If you want I can put them back in. However, I think we should rather come up with a true package-level comment instead of comments describing what the file does.

File: pkg/kubelet/metrics/metrics.go
Comment (obitech) [reply pkg/kubelet/metrics/metrics.go]: I was wondering about the same. I'll just use the help text for the comment then.

**Comment by obitech** [comment]
All tests are passing, assigning additional approver for `hack/`:

/assign @soltysh

**Comment by obitech** [comment]
@soltysh  oops, I think I mistakenly assigned you to this PR. Can I remove you again?

**Comment by derekwaynecarr** [comment]
I have one question on why the one import was removed.... otherwise this is good, and appreciate your help improving code quality.

**Comment by obitech** [comment]
@derekwaynecarr golint was throwing the following error:

```
pkg/kubelet/config/common.go:37:2: a blank import should be only in a main or test package, or have a comment justifying it
```

From what I can tell the import is not used anymore and I had troubles coming up with a reason/comment why it should remain, so I removed it.

**Comment by obitech** [comment]
Does this PR need any more attention from me? Any chance to merge this before the code freeze?

**Comment by mattjmcnaughton** [comment]
> Does this PR need any more attention from me? Any chance to merge this before the code freeze?

Unfortunately, I think this missed code freeze :( Hopefully it will be an early candidate for merge when the repo unfreezes.

In the mean time, I think you need to do a quick rebase because of some conflicts with `hack/.golint_failures`.

Thanks again for your work here :)

**Comment by obitech** [comment]
Ah no worries, I'll wait a bit longer then :)

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
have a question on why an import was removed.

File: pkg/kubelet/config/common.go
Comment (obitech) [reply pkg/kubelet/config/common.go]: I removed it becaused `golint` was complaining the blank import:

```
pkg/kubelet/config/common.go:37:2: a blank import should be only in a main or test package, or have a comment justifying it
```

Comment (obitech) [reply pkg/kubelet/config/common.go]: @derekwaynecarr should I put it back in? I'll have to put `pkg/kubelet/config` back on `.golint-failures` though.

Comment (derekwaynecarr) [reply pkg/kubelet/config/common.go]: that package has an init function.

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/install/install.go#L29

I think you need to keep the import but add a comment so it passes.

Comment (obitech) [reply pkg/kubelet/config/common.go]: I put the import back in, unfortunately `golint` doesn't seem to allow ignoring individual lines so I had to put `pkg/kubelet/config` back into `.golint-failures`. But we're still passing a lot more `golint` so this should be fine I think

**Review by derekwaynecarr — APPROVED** [review]
/approve
/lgtm

File: pkg/kubelet/config/common.go
Comment (BenTheElder) [reply pkg/kubelet/config/common.go]: > I think you need to keep the import but add a comment so it passes.

not a comment to suppress go lint or something, go lint is requiring that you have a comment justifying the import if you have an underscore import, IE we should comment about the dependency on `init()`

File: pkg/kubelet/config/apiserver.go
Comment (BenTheElder) [reply pkg/kubelet/config/apiserver.go]: I don't think we need this comment here, the methods in this file are commented.

the package itself should be documented though.

**Review by BenTheElder — APPROVED** [review]
/approve
for hack

I suspect you can get these other packages passing again and remove them as well by commenting on why we underscore import these other packages.

**Pull request merged**
